File: backend/src/main/java/com/fernando/zssn/persistence/entity/Survivor.java
package com.fernando.zssn.persistence.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.LazyCollection;
import org.hibernate.annotations.LazyCollectionOption;

import javax.persistence.*;
import java.util.List;

@Data
@Entity
@Table(name = "survivors")
@AllArgsConstructor
@NoArgsConstructor
public class Survivor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String surname;
    private Integer age;
    private Float latitude;
    private Float longitude;
    private Integer points;
    @OneToMany(mappedBy = "survivor", cascade = CascadeType.ALL)
    @LazyCollection(LazyCollectionOption.FALSE)
    private List<Item> items;
    private Integer infectedReports;
    private Boolean isInfected;

    @JsonManagedReference
    public List<Item> getItems(){
        return items;
    }

    public void addInfectedReport()
    {
        this.setInfectedReports(this.getInfectedReports()+1);
    }
}

File: app/app/src/main/java/com/fernando/zssn/data/model/Survivor.kt
package com.fernando.zssn.data.model

import android.os.Parcelable
import com.fernando.zssn.data.model.Item
import kotlinx.android.parcel.Parcelize
import kotlinx.android.parcel.RawValue

@Parcelize
data class Survivor(
    val age: Int,
    val id: Int,
    val infectedReports: Int,
    val isInfected: Boolean,
    val items: @RawValue List<Item>,
    val latitude: Double,
    val longitude: Double,
    val name: String,
    val points: Int,
    val surname: String

) : Parcelable